src/App.js

```diff
@@ -1,25 +1,25 @@
-import React from 'react';
-import logo from './logo.svg';
-import './App.css';
+import React from "react";
+import "./App.css";
+import { Switch, Route } from "react-router-dom";
+import MovieDetails from "./Components/MovieDetailsPage/MovieDetails";
+import MainPage from "./Components/MainPage/MainPage";
+import Header from "./Components/MainPage/Header/Header";
 
-function App() {
+const App = () => {
+  const APIKEY = "3aece1730ec57334758cdeb57c0d6adb";
   return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          <p>heeeellllloooo...</p>
-          <h1>hi</h1>
-          Learn React
-        </a>
-      </header>
-    </div>
+    <>
+      <Header title="MOVIE DATABASE" />
+      <Switch>
+        <Route exact path="/">
+          <MainPage APIKEY={APIKEY} />
```

Comment:
In order not to have to pass the API key to every component that needs it, you can:

1. Use [React Context](https://reactjs.org/docs/context.html) and get it in any component that needs it without explicitly passing through all the tree through props

2. Define a separate API layer where it would be pre-defined and then all your components when they nbeed to get something from API would use this layer instead. Basically, something like 

`api.js`
```js
const API_KEY = 'XXX';

export const searchMovies = async (query) => {
    const data = await fetch(
      `https://api.themoviedb.org/3/search/movie?api_key=${API_KEY}&query=${query}`
    );
    return await data.json();
}
```

And then in the components:
```jsx
import { searchMovies } from './api.js'
//...
const data = await searchMovies(query);
```

Reply:
I was trying to use context to achieve this result but ran into an error I don't know how to fix. When I try to import the context in from './App' I get an error saying this:
![Capture](https://user-images.githubusercontent.com/56635645/95109976-0224bf80-0746-11eb-8458-4352285c296f.PNG)

I have tried to change the name slightly like adding ".js" or "./src" which has helped in the past but I can not seem to figure this one out or find a solution anywhere

Reply:
You probably need to go back several levels, like `../../App`

Reply:
That worked perfectly, thanks! Opted for option 1 and used context to pass down the key